### ISSUE TITLE
Schedule conflicts

### ISSUE THREAD

**Issue by bryanharley** [trms/wp-cablecast#17]
If I update scheduled items in Cablecast that have already been synced to WordPress, you can end up with duplicate schedule items on WordPress.

For example, Show A is scheduled for 6pm in Cablecast. It gets synced to WordPress. Now I change the schedule in Cablecast so Show B is at 6pm.  When this syncs to WordPress it results in both Show A and Show B displaying on the schedule page at 6pm.

**Comment by raytiley** [comment]
https://github.com/raytiley/wp-cablecast/blob/master/includes/sync.php#L353

When an item is deleted from the schedule it should be returned as a deleted record and deleted from wordpress ^^

Does this only happen f the run is changed last minute? How often does your cron run to sync?

**Comment by bryanharley** [comment]
Yes, it's typically "day of" changes.  For example, we schedule our council meetings for all day, because we don't know when they'll end and they have been 12+ hours.  So when a meeting ends, we shorten the run and fill in the rest of the day. 

Cron runs every hour 7am to 10pm.

**Comment by bryanharley** [comment]
Huh, weird.  Cron runs every hour between 7am and 10pm.  So it should be deleting it.  But for example, look at 8pm: https://cmac.tv/channel/cmac-1/?schedule_date=2019-06-18

**Comment by raytiley** [comment]
My best guess is that on systems with multiple channels and lots of runs the sync logic for the schedule might be too naive

We basically try to load max 500 runs starting from two days ago. Might need to make this a bit smarter and page through results like when syncing shows.

https://github.com/raytiley/wp-cablecast/blob/master/includes/sync.php#L16-L18

**Comment by bryanharley** [comment]
The schedule seems to sync OK, except for in the scenerio I mentioned: 

Show A is scheduled for a specific time, and Show B replaces it in the schedule at the exact same time. Then the website displays both Show A and Show B at the same time. 

Example, look at 5pm: https://cmac.tv/channel/cmac-1/?schedule_date=2020-06-23

**Comment by bryanharley** [comment]
Fixed by the 2nd commit, https://github.com/trms/wp-cablecast/pull/31. Thanks!

**Comment by bryanharley** [comment]
For some reason I'm having this problem again. Some existing schedule items that have been deleted in Cablecast aren't being deleted in WordPress, and some new schedule items in Cablecast aren't being added to WordPress.

For example, schedule item id 568849 is in Cablecast, but not WordPress. Schedule item ID 564592 is flagged as deleted in Cablecast, but hasn't been deleted from WordPress.
